### PR TITLE
Refactor feedback page to use new form builder

### PR DESCRIPTION
**Pull request by tyom**
Align the structure with establishing patterns and get the benefit of improved error summary for free.

### Before
localhost_3001_support

### After
localhost_3001_support 2

### CODE REVIEW

File: src/apps/builders.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function buildFormWithErrors (form = {}, errorMessages = {}) {
   const messages = pick(pickBy(errorMessages), formFieldNames)
 
   const fieldLabels = form.children
-    .filter(x => Object.keys(errorMessages).includes(x.name))
+    .filter(x => Object.keys(messages).includes(x.name))
```
Comment (ztolley) [inline src/apps/builders.js]: I know you didn't name it but something more descriptive than x maybe?

Comment (tyom) [reply src/apps/builders.js]: I think in cases like this `x` as the name of the variable is somewhat irrelevant. In functional programming `x` is [often used as a variable name](http://blog.ploeh.dk/2015/08/17/when-x-y-and-z-are-great-variable-names/). It's a single generic parameter, a thing that is being operated on.

Comment (ztolley) [reply src/apps/builders.js]: cant hurt to have a sensible name rather than x

Comment (tyom) [reply src/apps/builders.js]: Did you read that article I linked? The scope is so small here spelling out full name of what it is unnecessary. I'm all for good variable names but in this case I think `x` works well.

Comment (ztolley) [reply src/apps/builders.js]: Yup I read it and yes I'm seen examples that just use x. Not going to let it stop this, just I like to use names.

File: src/apps/support/middleware.js [hunk 2]
```diff
@@ -1,100 +1,44 @@
...
-
-  res.locals.form = Object.assign({}, res.locals.form, {
-    errors,
+  res.locals.formErrors = Object.assign({}, res.locals.formErrors, {
+    messages,
   })
 
-  next()
-}
+  console.log(res.locals.formErrors)
```
Comment (teneightfive) [inline src/apps/support/middleware.js]: 🚨

File: src/apps/support/middleware.js [hunk 3]
```diff
@@ -1,100 +1,44 @@
...
 
-async function submitForm (req, res, next) {
-  const errorMessages = get(res.locals, 'form.errors.messages')
+  const hasErrors = !isEmpty(res.locals.formErrors.messages) || res.locals.formErrors.summary
```
Comment (teneightfive) [inline src/apps/support/middleware.js]: Should we use lodash's `get` to find these deep properties?

File: src/apps/support/middleware.js [hunk 4]
```diff
@@ -1,100 +1,44 @@
...
-      { id: config.zenBrowser, value: req.body.browser },
-      { id: config.zenService, value: 'datahub_export' },
-    ],
-    tags: [req.body.type],
-  }
+  const ticket = createZenDeskMessage(req.body)
 
   try {
     const response = await postToZenDesk(ticket)
     req.flash('success', `Created new report, reference number ${response.data.ticket.id}`)
```
Comment (teneightfive) [inline src/apps/support/middleware.js]: Can you stick a todo in here to display this in a better fashion on the complete page.

Would be better to use a pattern like [confirmation pages](https://www.gov.uk/service-manual/design/confirmation-pages) to play the reference back to the user.

File: src/apps/support/services.js [hunk 5]
```diff
@@ -2,6 +2,24 @@ const axios = require('axios')
...
 
+function createZenDeskMessage (body) {
+  return {
+    requester: {
+      name: 'Data Hub user',
+      email: body.email,
+    },
+    subject: body.title,
+    comment: {
+      body: body.description || 'N/A',
```
Comment (teneightfive) [inline src/apps/support/services.js]: Can we not just leave description empty? `N/A` feels like a hack to get round badly required fields.

File: src/apps/support/middleware.js (hunk 3 above)
Comment (tyom) [reply src/apps/support/middleware.js]: We're ensuring that `formErrors` object exists just a few lines above so we're confident to check properties on it.

File: src/apps/builders.js (hunk 1 above)
Comment (teneightfive) [reply src/apps/builders.js]: `stuff` :trollface:

**Pull request merged**